torch/csrc/jit/frontend/ir_emitter.cpp

```diff
@@ -3675,24 +3675,25 @@ struct to_ir {
         }
         AT_ASSERT(key_type != nullptr && value_type != nullptr);
 
-        auto checkTypeOfValues = [](const TypePtr& type,
-                                    const char* what,
-                                    const std::vector<Value*>& values,
-                                    TreeList trees) {
-          for (size_t i = 0, N = values.size(); i < N; ++i) {
-            std::stringstream ss;
-            if (!values[i]->type()->isSubtypeOfExt(type, &ss)) {
-              throw ErrorReport(trees[i])
-                  << "Dict " << what
-                  << " must contain only a single type, expected: "
-                  << type->repr_str() << " but found "
-                  << values[i]->type()->repr_str() << " instead.\n"
-                  << ss.str();
-            }
+        for (size_t i = 0, N = values.size(); i < N; ++i) {
+          std::stringstream ss;
+          if (!keys[i]->type()->isSubtypeOfExt(key_type, &ss) &&
+              !key_type->isSubtypeOfExt(keys[i]->type(), &ss)) {
```

Comment:
Same here, should `key_type` be updated to super type?

Reply:
This might be a good follow-up project, but it's a lot bigger than it seems on the surface. The reason we restrict the allowable keys to a small set of types is that we generate a unique schema for each type